static/src/javascripts/projects/common/modules/commercial/membership-engagement-banner.js

```diff
@@ -0,0 +1,215 @@
+    variantId: string
+): string => {
+    let prefix = '';
+    const offerings = membershipEngagementBannerUtils.offerings;
+
+    // mem and cont chosen to be consistent with default campaign code prefixes.
+    if (offering === offerings.membership) {
+        prefix = 'mem_';
+    } else if (offering === offerings.contributions) {
+        prefix = 'cont_';
+    }
+
+    return `${prefix + campaignId}_${variantId}`;
+};
+
+const getUserVariantParams = (
+    userVariant: ?Variant,
+    campaignId: ?string,
+    defaultOffering: string
+): Object => {
+    if (campaignId && userVariant && userVariant.engagementBannerParams) {
+        const userVariantParams: any = userVariant.engagementBannerParams;
+
+        if (!userVariantParams.campaignCode) {
+            const offering: string =
+                userVariantParams.offering || defaultOffering;
+            userVariantParams.campaignCode = buildCampaignCode(
+                offering,
+                campaignId,
+                userVariant.id
+            );
+        }
+
+        return userVariantParams;
+    }
+    return {};
+};
+
+/*
+ * Params for the banner are overlaid in this order, earliest taking precedence:
+ *
+ *  * Variant (if the user is in an A/B testing variant)
+ *  * Edition
+ *  * Offering ('membership' or 'contributions')
+ *  * Default
+ *
+ * The 'offering' in use comes from either:
+ *
+ *  * Variant (if the user is in an A/B testing variant)
+ *  * Edition (only one offering can be the default for a given Edition)
+ *
+ * Returns either 'null' if no banner is available for this edition,
+ * otherwise a populated params object that looks like this:
+ *
+ *  {
+ *    minArticles: 5, // how many articles should the user see before they get the engagement banner?
+ *    messageText: "..."
+ *    colourStrategy: // a function to determine what css class to use for the banner's colour
+ *    buttonCaption: "Become a Supporter"
+ *  }
+ *
+ */
+const deriveBannerParams = (location: string): Object | string => {
+    const defaultParams = membershipEngagementBannerUtils.defaultParams(
+        location
+    );
+    const userTest = getUserTest();
+    const campaignId = userTest ? userTest.campaignId : undefined;
+    const userVariant = getUserVariant(userTest);
+
+    if (userVariant && userVariant.blockEngagementBanner) {
+        return DO_NOT_RENDER_ENGAGEMENT_BANNER;
+    }
+
+    return Object.assign(
+        {},
+        defaultParams,
+        getUserVariantParams(userVariant, campaignId, defaultParams.offering)
+    );
+};
+
+// Used to send an interaction if the engagement banner is shown.
+const recordInteraction = (interaction: Object): void => {
+    if (interaction) {
+        const component = interaction.component;
+        const value = interaction.value;
+
+        if (component && value) {
+            ophan.record({
+                component,
+                value,
+            });
+        }
+    }
+};
+
+const paypalAndCreditCardImage =
```

Comment:
Could you move this to the top of the file since it's only a static assignment? Could also be a good usage of

```js
config.get('images.acquisitions.paypal-and-credit-card', '')
```